tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/amqp/AmqpFlowControlFailDispositionTests.java

```diff
@@ -105,4 +108,51 @@ public void testAddressFullDisposition() throws Exception {
          connection.close();
       }
    }
+
+   @TestTemplate
+   @Timeout(60)
+   public void testFailedLargeMessageSendWhenNoSpaceCleansUpLargeFile() throws Exception {
+      AmqpClient client = createAmqpClient(getBrokerAmqpConnectionURI());
+      AmqpConnection connection = client.connect();
+
+      int expectedRemainingLargeMessageFiles = 0;
+
+      try {
+         AmqpSession session = connection.createSession();
+         AmqpSender sender = session.createSender(getQueueName(), null, null, outcomes);
+         AmqpMessage message = createAmqpLargeMessageWithNoBody();
+         boolean rejected = false;
+
+         for (int i = 0; i < 1000; i++) {
+            try {
+               sender.send(message);
+               expectedRemainingLargeMessageFiles++;
+            } catch (IOException e) {
+               rejected = true;
+               assertTrue(e.getMessage().contains(expectedMessage),
+                          String.format("Unexpected message expected %s to contain %s", e.getMessage(), expectedMessage));
+               break;
+            }
+         }
+
+         assertTrue(rejected, "Expected messages to be refused by broker");
+      } finally {
+         connection.close();
+      }
+
+      validateNoFilesOnLargeDir(getLargeMessagesDir(), expectedRemainingLargeMessageFiles);
+   }
+
+   private AmqpMessage createAmqpLargeMessageWithNoBody() {
+      AmqpMessage message = new AmqpMessage();
+
+      byte[] payload = new byte[32 * 1024];
```

Comment:
Would be good to tie this to the constant size so that it remains 'large' in the case anyone ever changed that config (e.g as you just did) without also updating this bit....test would still pass, but no longer cover what was intended.